src/main/java/com/google/devtools/build/lib/syntax/MethodDescriptor.java

```diff
@@ -122,48 +263,19 @@ Object callField(Object obj, StarlarkSemantics semantics, @Nullable Mutability m
    *
    * <p>The Mutability is used if it is necessary to allocate a Starlark copy of a Java result.
    */
```

Comment:
Document that thread must be non-null iff this is a method, not a structField.

Reply:
I have rewritten this code to create different method handles for fields and other methods.